handler/schedule.go

```diff
@@ -155,25 +170,41 @@ func (s *ScheduleHandler) updateStatus() error {
 	return nil
 }
 
-func (s *ScheduleHandler) getEffectiveSchedule(jobType k8upv1alpha1.JobType, originalSchedule string) string {
+func (s *ScheduleHandler) getEffectiveSchedule(jobType k8upv1alpha1.JobType, originalSchedule k8upv1alpha1.ScheduleDefinition) k8upv1alpha1.ScheduleDefinition {
 	if s.schedule.Status.EffectiveSchedules == nil {
-		s.schedule.Status.EffectiveSchedules = make(map[k8upv1alpha1.JobType]string)
+		s.schedule.Status.EffectiveSchedules = make(map[k8upv1alpha1.JobType]k8upv1alpha1.ScheduleDefinition)
 	}
 	if existingSchedule, found := s.schedule.Status.EffectiveSchedules[jobType]; found {
 		return existingSchedule
 	}
-	if !strings.HasSuffix(originalSchedule, "-random") {
+	if !originalSchedule.IsNonStandard() {
 		return originalSchedule
 	}
-	schedule := originalSchedule
+	if originalSchedule.IsRandom() {
+		randomizedSchedule, err := s.getRandomSchedule(jobType, originalSchedule)
+		if err == nil {
+			s.setEffectiveSchedule(jobType, randomizedSchedule)
+			return randomizedSchedule
+		}
+	}
+
+	s.setEffectiveSchedule(jobType, originalSchedule)
+	return originalSchedule
+}
+
+func (s *ScheduleHandler) getRandomSchedule(jobType k8upv1alpha1.JobType, originalSchedule k8upv1alpha1.ScheduleDefinition) (k8upv1alpha1.ScheduleDefinition, error) {
 	seed := s.createSeed(s.schedule, jobType)
-	schedule, err := randomizeSchedule(seed, originalSchedule)
+	randomizedSchedule, err := randomizeSchedule(seed, originalSchedule)
 	if err != nil {
-		s.Log.Info("Could not randomize schedule, ignoring and try original schedule", "error", err.Error())
-	} else {
-		s.Log.V(1).Info("Randomized schedule", "seed", seed, "from_schedule", originalSchedule, "effective_schedule", schedule)
+		s.Log.Info("Could not randomize schedule", "schedule", originalSchedule, "error", err.Error())
```

Comment:
I propose to move this log line to the caller in `getEffectiveSchedule`, so that we follow better the "either-log-or-throw-but-not-both" paradigma.